spyder/plugins/workingdirectory/plugin.py

```diff
@@ -4,232 +4,165 @@
-
-        # Parent dir action
-        parent_action = create_action(self, "parent", None,
-                                      ima.icon('up'),
-                                      _('Change to parent directory'),
-                                      triggered=self.parent_directory)
-        self.toolbar.addAction(parent_action)
+    # --- Signals
+    # ------------------------------------------------------------------------
+    sig_next_enabled = Signal(bool)
+    """
+    """
 
-    def get_workdir(self):
-        """Get current workdir from the CONF file."""
-        if self.get_option('startup/use_fixed_directory'):
-            workdir = self.get_option('startup/fixed_directory',
-                                      default='')
-        elif self.get_option('console/use_project_or_home_directory'):
-            workdir = get_home_dir()
-        else:
-            workdir = self.get_option('console/fixed_directory',
-                                      default='')
-        if not osp.isdir(workdir):
-            workdir = get_home_dir()
-        return workdir
-
-    #------ SpyderPluginWidget API ---------------------------------------------    
-    def get_plugin_title(self):
-        """Return widget title"""
+    sig_previous_enabled = Signal(bool)
+    """
+    """
+
+    sig_current_directory_changed = Signal(str)
+    """
+    """
+
+    # --- SpyderPluginV2 API
+    # ------------------------------------------------------------------------
+    def get_name(self):
         return _('Current working directory')
-    
-    def get_plugin_icon(self):
-        """Return widget icon"""
-        return ima.icon('DirOpenIcon')
-
-    def register_plugin(self):
-        """Register plugin in Spyder's main window"""
-        self.redirect_stdio.connect(self.main.redirect_internalshell_stdio)
-        self.main.console.sig_refreshed.connect(self.refresh_plugin)
-        iconsize = 24
-        self.toolbar.setIconSize(QSize(iconsize, iconsize))
-        self.main.addToolBar(self.toolbar)
-        
-    def refresh_plugin(self):
-        """Refresh widget"""
-        curdir = getcwd_or_home()
-        self.pathedit.add_text(curdir)
-        self.save_wdhistory()
-        self.set_previous_enabled.emit(
-                             self.histindex is not None and self.histindex > 0)
-        self.set_next_enabled.emit(self.histindex is not None and \
-                                   self.histindex < len(self.history)-1)
-
-    #------ Public API ---------------------------------------------------------
-    def load_wdhistory(self, workdir=None):
-        """Load history from a text file in user home directory"""
-        if osp.isfile(self.LOG_PATH):
-            wdhistory, _ = encoding.readlines(self.LOG_PATH)
-            wdhistory = [name for name in wdhistory if os.path.isdir(name)]
+
+    def get_description(self):
+        return _('Set current working directory on different spyder plugins')

```

Comment:
Please change this to

    Set the current working directory for various plugins.

Reply:
Fixed